packages/p2p/src/components/modal-manager/modal-manager-context-provider.jsx

```diff
@@ -9,8 +9,31 @@ const ModalManagerContextProvider = props => {
     // for mobile, modals are stacked and not shown alternatingly one by one
     const [stacked_modal, setStackedModal] = React.useState({});
     const [is_modal_open, setIsModalOpen] = React.useState(false);
+    const [modal_props, setModalProps] = React.useState(new Map());
     const { general_store } = useStores();
 
+    /**
+     * Sets the specified modals' props on mount or when the props passed to the hook has changed.
+     *
+     * Use this hook to declare the modals' props beforehand for cases when the props can't be passed/declared in stores.
+     *
+     * For instance, calling `showModal({key: ..., props: ... })` in a store action where the props can't be passed to the action, use this hook to pass the props beforehand
+     * and simply call `showModal({key: ...})` without the need to specify the props, since its already passed using this hook to the modal manager.
+     *
+     * @param {Object|Object[]} modals - list of object modals to set props, each modal object must contain a 'key' attribute and 'props' attribute
+     */
+    const useRegisterModalProps = modals => {
+        const registerModals = React.useCallback(() => {
+            if (Array.isArray(modals)) {
+                modals.forEach(modal => setModalProps(modal_props.set(modal.key, modal.props)));
+            } else {
+                setModalProps(modal_props.set(modals.key, modals.props));
+            }
+        }, [modals]);
+
+        React.useEffect(registerModals, [modals]);
+    };
```

Comment:
The reason I created this hook is for the flexibility of passing local state props for situations where the local state props can't be passed to the `showModal`, especially in stores.
We could just pass the props through the action arguments, but then if the actions are nested within another action, it'll just create another prop drilling in the form of arguments.